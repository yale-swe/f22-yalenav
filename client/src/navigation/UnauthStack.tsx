import SignInScreen from "../components/auth/SignInScreen";
import HomeScreen from "../screens/HomeScreen";
import { createStackNavigator } from "@react-navigation/stack";
import { RootStackParamList } from "./Navigation";

export const Stack = createStackNavigator<RootStackParamList>();

export default function UnauthStack() {
  return (
    <Stack.Navigator
      initialRouteName="Home"
      screenOptions={{ headerShown: false }}
    >
      <Stack.Screen name="Home" component={HomeScreen} />
      <Stack.Screen name="SignIn" component={SignInScreen} />
    </Stack.Navigator>
  );
}
