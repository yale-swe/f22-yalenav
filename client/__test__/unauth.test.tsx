import { NavigationContainer } from "@react-navigation/native";
import UnauthStack from "../src/navigation/UnauthStack";
import renderer from "react-test-renderer";

describe("Testing not logged in", () => {
  test("Renders Correctly (integration test)", async () => {
    const component = (
      <NavigationContainer>
        <UnauthStack />
      </NavigationContainer>
    );
    const tree = renderer.create(component).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
