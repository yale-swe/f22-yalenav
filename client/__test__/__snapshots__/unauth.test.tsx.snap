// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing not logged in Renders Correctly (integration test) 1`] = `
<View
  style={
    {
      "flex": 1,
    }
  }
>
  <RNCSafeAreaProvider
    onInsetsChange={[Function]}
    style={
      [
        {
          "flex": 1,
        },
        undefined,
      ]
    }
  >
    <View
      style={
        [
          {
            "backgroundColor": "rgb(242, 242, 242)",
            "flex": 1,
          },
          undefined,
        ]
      }
    >
      <RNSScreenContainer
        onLayout={[Function]}
        style={
          {
            "flex": 1,
          }
        }
      >
        <RNSScreen
          activityState={2}
          collapsable={false}
          forwardedRef={[Function]}
          gestureResponseDistance={
            {
              "bottom": -1,
              "end": -1,
              "start": -1,
              "top": -1,
            }
          }
          pointerEvents="box-none"
          style={
            {
              "bottom": 0,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            }
          }
        >
          <View
            collapsable={false}
            style={
              {
                "opacity": 1,
              }
            }
          />
          <View
            accessibilityElementsHidden={false}
            closing={false}
            gestureVelocityImpact={0.3}
            importantForAccessibility="auto"
            onClose={[Function]}
            onGestureBegin={[Function]}
            onGestureCanceled={[Function]}
            onGestureEnd={[Function]}
            onOpen={[Function]}
            onTransition={[Function]}
            pointerEvents="box-none"
            style={
              [
                {
                  "display": "flex",
                  "overflow": undefined,
                },
                {
                  "bottom": 0,
                  "left": 0,
                  "position": "absolute",
                  "right": 0,
                  "top": 0,
                },
              ]
            }
            transitionSpec={
              {
                "close": {
                  "animation": "spring",
                  "config": {
                    "damping": 500,
                    "mass": 3,
                    "overshootClamping": true,
                    "restDisplacementThreshold": 10,
                    "restSpeedThreshold": 10,
                    "stiffness": 1000,
                  },
                },
                "open": {
                  "animation": "spring",
                  "config": {
                    "damping": 500,
                    "mass": 3,
                    "overshootClamping": true,
                    "restDisplacementThreshold": 10,
                    "restSpeedThreshold": 10,
                    "stiffness": 1000,
                  },
                },
              }
            }
          >
            <View
              collapsable={false}
              pointerEvents="box-none"
              style={
                {
                  "flex": 1,
                  "marginTop": 0,
                }
              }
            >
              <View
                collapsable={false}
                forwardedRef={[Function]}
                handlerTag={1}
                handlerType="PanGestureHandler"
                needsOffscreenAlphaCompositing={false}
                onGestureHandlerEvent={[Function]}
                onGestureHandlerStateChange={[Function]}
                style={
                  {
                    "flex": 1,
                    "transform": [
                      {
                        "translateX": 0,
                      },
                      {
                        "translateX": 0,
                      },
                    ],
                  }
                }
              >
                <View
                  pointerEvents="box-none"
                  style={
                    [
                      {
                        "flex": 1,
                        "overflow": "hidden",
                      },
                      [
                        {
                          "backgroundColor": "rgb(242, 242, 242)",
                        },
                        undefined,
                      ],
                    ]
                  }
                >
                  <View
                    style={
                      {
                        "flex": 1,
                        "flexDirection": "column-reverse",
                      }
                    }
                  >
                    <View
                      style={
                        {
                          "flex": 1,
                        }
                      }
                    >
                      <MapView
                        customMapStyle={
                          [
                            {
                              "elementType": "geometry",
                              "stylers": [
                                {
                                  "color": "#f5f5f5",
                                },
                              ],
                            },
                            {
                              "elementType": "labels.icon",
                              "stylers": [
                                {
                                  "visibility": "off",
                                },
                              ],
                            },
                            {
                              "elementType": "labels.text.fill",
                              "stylers": [
                                {
                                  "color": "#616161",
                                },
                              ],
                            },
                            {
                              "elementType": "labels.text.stroke",
                              "stylers": [
                                {
                                  "color": "#f5f5f5",
                                },
                              ],
                            },
                            {
                              "elementType": "labels.text.fill",
                              "featureType": "administrative.land_parcel",
                              "stylers": [
                                {
                                  "color": "#bdbdbd",
                                },
                              ],
                            },
                            {
                              "elementType": "labels.text",
                              "featureType": "administrative.locality",
                              "stylers": [
                                {
                                  "visibility": "off",
                                },
                              ],
                            },
                            {
                              "elementType": "labels.text",
                              "featureType": "administrative.neighborhood",
                              "stylers": [
                                {
                                  "visibility": "off",
                                },
                              ],
                            },
                            {
                              "elementType": "labels.text",
                              "featureType": "administrative.province",
                              "stylers": [
                                {
                                  "visibility": "off",
                                },
                              ],
                            },
                            {
                              "elementType": "geometry",
                              "featureType": "poi",
                              "stylers": [
                                {
                                  "color": "#eeeeee",
                                },
                              ],
                            },
                            {
                              "elementType": "labels.text.fill",
                              "featureType": "poi",
                              "stylers": [
                                {
                                  "color": "#757575",
                                },
                              ],
                            },
                            {
                              "elementType": "geometry",
                              "featureType": "poi.park",
                              "stylers": [
                                {
                                  "color": "#e5e5e5",
                                },
                              ],
                            },
                            {
                              "elementType": "labels.text.fill",
                              "featureType": "poi.park",
                              "stylers": [
                                {
                                  "color": "#9e9e9e",
                                },
                              ],
                            },
                            {
                              "elementType": "geometry",
                              "featureType": "road",
                              "stylers": [
                                {
                                  "color": "#ffffff",
                                },
                              ],
                            },
                            {
                              "elementType": "labels.text.fill",
                              "featureType": "road.arterial",
                              "stylers": [
                                {
                                  "color": "#757575",
                                },
                              ],
                            },
                            {
                              "elementType": "geometry",
                              "featureType": "road.highway",
                              "stylers": [
                                {
                                  "color": "#dadada",
                                },
                              ],
                            },
                            {
                              "elementType": "labels.text.fill",
                              "featureType": "road.highway",
                              "stylers": [
                                {
                                  "color": "#616161",
                                },
                              ],
                            },
                            {
                              "elementType": "labels.text.fill",
                              "featureType": "road.local",
                              "stylers": [
                                {
                                  "color": "#9e9e9e",
                                },
                              ],
                            },
                            {
                              "elementType": "geometry",
                              "featureType": "transit.line",
                              "stylers": [
                                {
                                  "color": "#e5e5e5",
                                },
                              ],
                            },
                            {
                              "elementType": "geometry",
                              "featureType": "transit.station",
                              "stylers": [
                                {
                                  "color": "#eeeeee",
                                },
                              ],
                            },
                            {
                              "elementType": "geometry",
                              "featureType": "water",
                              "stylers": [
                                {
                                  "color": "#c9c9c9",
                                },
                              ],
                            },
                            {
                              "elementType": "labels.text.fill",
                              "featureType": "water",
                              "stylers": [
                                {
                                  "color": "#9e9e9e",
                                },
                              ],
                            },
                          ]
                        }
                        followsUserLocation={true}
                        initialRegion={
                          {
                            "latitude": 41.3163,
                            "latitudeDelta": 0.0622,
                            "longitude": -72.922585,
                            "longitudeDelta": 0.0121,
                          }
                        }
                        showsUserLocation={true}
                        style={
                          {
                            "alignSelf": "stretch",
                            "height": "125%",
                          }
                        }
                      />
                      <View
                        style={
                          {
                            "flex": 1,
                            "flexDirection": "row",
                            "justifyContent": "space-around",
                            "paddingTop": "12%",
                            "position": "absolute",
                          }
                        }
                      >
                        <View>
                          <View
                            style={
                              {
                                "flex": 1,
                                "padding": "2%",
                                "paddingLeft": "4%",
                              }
                            }
                          >
                            <View
                              collapsable={false}
                              style={
                                {
                                  "alignItems": "center",
                                  "backgroundColor": "white",
                                  "borderColor": "#00356B",
                                  "borderRadius": 40,
                                  "borderWidth": 2,
                                  "elevation": 0,
                                  "flexDirection": "row",
                                }
                              }
                            >
                              <View
                                accessibilityLabel="search"
                                accessibilityRole="button"
                                accessibilityState={
                                  {
                                    "disabled": true,
                                  }
                                }
                                accessible={true}
                                focusable={false}
                                hitSlop={
                                  {
                                    "bottom": 6,
                                    "left": 6,
                                    "right": 6,
                                    "top": 6,
                                  }
                                }
                                onClick={[Function]}
                                onResponderGrant={[Function]}
                                onResponderMove={[Function]}
                                onResponderRelease={[Function]}
                                onResponderTerminate={[Function]}
                                onResponderTerminationRequest={[Function]}
                                onStartShouldSetResponder={[Function]}
                                style={
                                  [
                                    {
                                      "overflow": "hidden",
                                    },
                                    [
                                      {
                                        "alignItems": "center",
                                        "justifyContent": "center",
                                        "margin": 6,
                                        "overflow": "hidden",
                                      },
                                      {
                                        "borderRadius": 18,
                                        "height": 36,
                                        "width": 36,
                                      },
                                      undefined,
                                      undefined,
                                    ],
                                  ]
                                }
                              >
                                <View>
                                  <Text />
                                </View>
                              </View>
                              <TextInput
                                accessibilityRole="search"
                                accessibilityTraits="search"
                                autoCorrect={false}
                                keyboardAppearance="light"
                                onChangeText={[Function]}
                                placeholder="Search for a Yale Location..."
                                placeholderTextColor="rgba(0, 0, 0, 0.54)"
                                returnKeyType="search"
                                selectionColor="#6200ee"
                                style={
                                  [
                                    {
                                      "alignSelf": "stretch",
                                      "flex": 1,
                                      "fontSize": 18,
                                      "minWidth": 0,
                                      "paddingLeft": 8,
                                      "textAlign": "left",
                                    },
                                    {
                                      "color": "#000000",
                                      "fontFamily": "System",
                                      "fontWeight": "400",
                                    },
                                    {
                                      "fontSize": 13,
                                    },
                                  ]
                                }
                                underlineColorAndroid="transparent"
                                value=""
                              />
                              <View
                                accessibilityLabel="clear"
                                accessibilityRole="button"
                                accessibilityState={
                                  {
                                    "disabled": true,
                                  }
                                }
                                accessible={true}
                                focusable={true}
                                hitSlop={
                                  {
                                    "bottom": 6,
                                    "left": 6,
                                    "right": 6,
                                    "top": 6,
                                  }
                                }
                                onClick={[Function]}
                                onResponderGrant={[Function]}
                                onResponderMove={[Function]}
                                onResponderRelease={[Function]}
                                onResponderTerminate={[Function]}
                                onResponderTerminationRequest={[Function]}
                                onStartShouldSetResponder={[Function]}
                                style={
                                  [
                                    {
                                      "overflow": "hidden",
                                    },
                                    [
                                      {
                                        "alignItems": "center",
                                        "justifyContent": "center",
                                        "margin": 6,
                                        "overflow": "hidden",
                                      },
                                      {
                                        "borderRadius": 18,
                                        "height": 36,
                                        "width": 36,
                                      },
                                      {
                                        "opacity": 0.32,
                                      },
                                      undefined,
                                    ],
                                  ]
                                }
                              >
                                <View>
                                  <Text />
                                </View>
                              </View>
                            </View>
                          </View>
                          <View
                            style={
                              {
                                "alignSelf": "center",
                              }
                            }
                          >
                            <View
                              style={
                                {
                                  "flexDirection": "row",
                                  "justifyContent": "space-around",
                                  "width": "80%",
                                }
                              }
                            >
                              <View
                                accessible={true}
                                collapsable={false}
                                focusable={true}
                                onClick={[Function]}
                                onResponderGrant={[Function]}
                                onResponderMove={[Function]}
                                onResponderRelease={[Function]}
                                onResponderTerminate={[Function]}
                                onResponderTerminationRequest={[Function]}
                                onStartShouldSetResponder={[Function]}
                                style={
                                  {
                                    "backgroundColor": "white",
                                    "borderColor": "#00356B",
                                    "borderRadius": 40,
                                    "borderWidth": 2,
                                    "opacity": 1,
                                    "padding": 7,
                                  }
                                }
                              >
                                <Text
                                  style={
                                    [
                                      {
                                        "alignSelf": "center",
                                        "color": "#00356B",
                                      },
                                      null,
                                    ]
                                  }
                                >
                                  Colleges
                                </Text>
                              </View>
                              <View
                                accessible={true}
                                collapsable={false}
                                focusable={true}
                                onClick={[Function]}
                                onResponderGrant={[Function]}
                                onResponderMove={[Function]}
                                onResponderRelease={[Function]}
                                onResponderTerminate={[Function]}
                                onResponderTerminationRequest={[Function]}
                                onStartShouldSetResponder={[Function]}
                                style={
                                  {
                                    "backgroundColor": "white",
                                    "borderColor": "#00356B",
                                    "borderRadius": 40,
                                    "borderWidth": 2,
                                    "opacity": 1,
                                    "padding": 7,
                                  }
                                }
                              >
                                <Text
                                  style={
                                    [
                                      {
                                        "alignSelf": "center",
                                        "color": "#00356B",
                                      },
                                      null,
                                    ]
                                  }
                                >
                                  Dining Halls
                                </Text>
                              </View>
                              <View
                                accessible={true}
                                collapsable={false}
                                focusable={true}
                                onClick={[Function]}
                                onResponderGrant={[Function]}
                                onResponderMove={[Function]}
                                onResponderRelease={[Function]}
                                onResponderTerminate={[Function]}
                                onResponderTerminationRequest={[Function]}
                                onStartShouldSetResponder={[Function]}
                                style={
                                  {
                                    "backgroundColor": "white",
                                    "borderColor": "#00356B",
                                    "borderRadius": 40,
                                    "borderWidth": 2,
                                    "opacity": 1,
                                    "padding": 7,
                                  }
                                }
                              >
                                <Text
                                  style={
                                    [
                                      {
                                        "alignSelf": "center",
                                        "color": "#00356B",
                                      },
                                      null,
                                    ]
                                  }
                                >
                                  Libraries
                                </Text>
                              </View>
                            </View>
                          </View>
                        </View>
                        <View
                          style={
                            {
                              "padding": "2%",
                              "paddingRight": "4%",
                            }
                          }
                        >
                          <View
                            style={
                              [
                                {
                                  "borderRadius": 3,
                                  "overflow": "hidden",
                                },
                                {
                                  "borderRadius": 3,
                                },
                                undefined,
                                false,
                              ]
                            }
                          >
                            <View
                              accessibilityRole="button"
                              accessibilityState={
                                {
                                  "busy": false,
                                  "disabled": false,
                                }
                              }
                              accessible={true}
                              collapsable={false}
                              focusable={true}
                              onClick={[Function]}
                              onResponderGrant={[Function]}
                              onResponderMove={[Function]}
                              onResponderRelease={[Function]}
                              onResponderTerminate={[Function]}
                              onResponderTerminationRequest={[Function]}
                              onStartShouldSetResponder={[Function]}
                              style={
                                {
                                  "backgroundColor": "white",
                                  "borderColor": "#00356B",
                                  "borderRadius": 40,
                                  "borderWidth": 2,
                                  "opacity": 1,
                                }
                              }
                            >
                              <View
                                style={
                                  {
                                    "alignItems": "center",
                                    "backgroundColor": "transparent",
                                    "borderColor": "#2089dc",
                                    "borderRadius": 3,
                                    "borderWidth": 0,
                                    "flexDirection": "row",
                                    "justifyContent": "center",
                                    "padding": 8,
                                  }
                                }
                              >
                                <Text
                                  style={
                                    {
                                      "color": "#2089dc",
                                      "fontSize": 18,
                                      "paddingVertical": 1,
                                      "textAlign": "center",
                                    }
                                  }
                                >
                                  Sign In
                                </Text>
                              </View>
                            </View>
                          </View>
                        </View>
                      </View>
                      <View
                        style={
                          {
                            "alignSelf": "center",
                            "backgroundColor": "white",
                            "borderColor": "#00356B",
                            "borderWidth": 2,
                            "flex": 1,
                            "height": "100%",
                            "marginTop": "197%",
                            "position": "absolute",
                            "shadowColor": "black",
                            "shadowRadius": 20,
                            "width": "120%",
                            "zIndex": 999,
                          }
                        }
                      >
                        <View
                          style={
                            [
                              {
                                "borderRadius": 3,
                                "overflow": "hidden",
                              },
                              {
                                "borderRadius": 3,
                              },
                              undefined,
                              false,
                            ]
                          }
                        >
                          <View
                            accessibilityRole="button"
                            accessibilityState={
                              {
                                "busy": false,
                                "disabled": false,
                              }
                            }
                            accessible={true}
                            collapsable={false}
                            focusable={true}
                            onClick={[Function]}
                            onResponderGrant={[Function]}
                            onResponderMove={[Function]}
                            onResponderRelease={[Function]}
                            onResponderTerminate={[Function]}
                            onResponderTerminationRequest={[Function]}
                            onStartShouldSetResponder={[Function]}
                            style={
                              {
                                "backgroundColor": "white",
                                "opacity": 1,
                                "padding": "2%",
                              }
                            }
                            testID="NextClassButton"
                          >
                            <View
                              style={
                                {
                                  "alignItems": "center",
                                  "backgroundColor": "transparent",
                                  "borderColor": "#2089dc",
                                  "borderRadius": 3,
                                  "borderWidth": 0,
                                  "flexDirection": "row",
                                  "justifyContent": "center",
                                  "padding": 8,
                                }
                              }
                            >
                              <Text />
                            </View>
                          </View>
                        </View>
                      </View>
                    </View>
                  </View>
                </View>
              </View>
            </View>
          </View>
        </RNSScreen>
      </RNSScreenContainer>
      <View
        collapsable={false}
        pointerEvents="box-none"
        style={
          {
            "height": 44,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
            "zIndex": 1,
          }
        }
      />
    </View>
  </RNCSafeAreaProvider>
</View>
`;
